metadata-ingestion/src/datahub/ingestion/source/cassandra/cassandra_api.py

```diff
@@ -0,0 +1,175 @@
+import logging
+from ssl import CERT_NONE, PROTOCOL_TLSv1_2, SSLContext
+from typing import List
+
+from cassandra.auth import PlainTextAuthProvider
+from cassandra.cluster import (
+    EXEC_PROFILE_DEFAULT,
+    Cluster,
+    ExecutionProfile,
+    ProtocolVersion,
+    Session,
+)
+
+from datahub.ingestion.api.source import SourceReport
+from datahub.ingestion.source.cassandra.cassandra_config import CassandraSourceConfig
+from datahub.ingestion.source.cassandra.cassandra_utils import (
+    CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES,
+    CassandraQueries,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class CassandraAPIInterface:
+    def __init__(self, config: CassandraSourceConfig, report: SourceReport):
+        self.config = config
+        self.report = report
+        self.cassandra_session = self.authenticate()
+
+    def authenticate(self) -> Session:
+        """Establish a connection to Cassandra and return the session."""
+        try:
+            if self.config.cloud:
+                cloud_config = self.config.cloud_config
+                assert cloud_config
```

Comment:
Can replace this check as `if self.config.cloud_config` if we remove empty defaults in config as suggested earlier.

Reply:
We have additional optional config so this check invalid right? self.config.cloud validated token and bundle path 
```
    connect_timeout: int = Field(
        default=60,
        description="Timeout in seconds for establishing new connections to Cassandra.",
    )

    request_timeout: int = Field(
        default=60, description="Timeout in seconds for individual Cassandra requests."
    )
```

Reply:
Presence of cloud_config mandates presence of secure_bundle and token if cloud_config is present due to pydantic validation, making these checks equivalent.